### DISCUSSION TITLE
xDrip accessibility permission reset after every xDrip update

### DISCUSSION THREAD

**Discussion by janvitos** [NightscoutFoundation/xDrip#3203] [Bug reports]
Hi,

I update xDrip every time there is a new nightly build, and each time, the xDrip accessibility permission in Android 14 is reset and Always On Display (AOD) no longer works. I then have to give xDrip the accessibility permission again.

To be honest, giving the permission is not what's bothering me the most. I use my phone on my desk as blood sugar monitor for my diabetic child, and when this happens, my phone doesn't show his blood sugar anymore, so it makes the feature kind of unreliable.

Is there a reason why this happens? Would there be a way to give xDrip this permission permanently even after updating?

Thanks!

**Comment by Navid200** [comment]
I experience the same.
It's not even limited to xDrip updates.  It happens sometimes even when there is no update.

**Reply by janvitos** [reply]
So you think this is a bug / feature in Android rather than xDrip?

**Reply by Navid200** [reply]
I don't know why it happens.  
Thanks for bringing it up.  
Let's see what @jamorham thinks.

**Comment by janvitos** [comment]
After doing a bit of research, I found out the "feature" was introduced in Android 13. Other app developers are experiencing the same issue, but there seems to be no fix:

https://stackoverflow.com/questions/77324695/how-to-enable-android-14-accessibility-restricted-setting

https://support.google.com/googleplay/android-developer/thread/239870179/android-14-app-accessibility-setting?hl=en

This is once again some nasty work by Google developers.

**Reply by Navid200** [reply]
Thanks for sharing.

**Reply by jamorham** [reply]
yes this is very likely the issue. I may be able to work around it but its complicated and potentially ugly which is why I haven't progressed much further with that.
